core/src/main/java/org/apache/calcite/sql/SqlIterationStmt.java

```diff
@@ -58,7 +58,7 @@ protected SqlIterationStmt(final SqlParserPos pos, final SqlNode condition,
       throw SqlUtil.newContextException(beginLabel.getParserPosition(),
           RESOURCE.beginEndLabelMismatch());
     }
-    this.condition = Objects.requireNonNull(condition);
+    this.condition = condition;
```

Comment:
nit: Update the comment to match that this can be null

Reply:
Good catch, fixed in https://github.com/googleinterns/calcite/pull/287/commits/5ec39d0522cb6f9f398b68a2482a2f18ed2ad4fc